tools/scripts/build/build-curriculum.js

```diff
@@ -2,11 +2,15 @@ const fs = require('fs');
 const path = require('path');
 
 const { getChallengesForLang } = require('../../../curriculum/getChallenges');
+const { buildMobileCurriculum } = require('./build-mobile-curriculum');
 
 const globalConfigPath = path.resolve(__dirname, '../../../config');
 
 // We are defaulting to English because the ids for the challenges are same
 // across all languages.
 getChallengesForLang('english')
   .then(JSON.stringify)
-  .then(x => fs.writeFileSync(`${globalConfigPath}/curriculum.json`, x));
+  .then(json => {
+    fs.writeFileSync(`${globalConfigPath}/curriculum.json`, json);
+    buildMobileCurriculum(json);
```

Comment:
Can't you avoid parsing the curriculum string by calling `buildMobileCurriculum` after `getChallengesForLang`?

Reply:
Not exactly sure what you mean by this.

Reply:
The current approach takes an object, stringifies it, writes it to a file and then has to parse that string again.  If we call `buildMobileCurriculum` before `JSON.stringify` we don't have to parse the string.

Reply:
This makes sense